### ISSUE TITLE
Question: Best way to handle S608 triggers on GAQL

### ISSUE THREAD

**Issue by lociii** [astral-sh/ruff#16156]
### Description

Hey there,

we're using the fantastic ruff tool to keep our Python projects in shape. Thanks for all the hard work that went into it. \

ruff version in use: `0.9.6`

As we're communicating with Google Ads APIs, we run into the problem that [GAQL](https://developers.google.com/google-ads/api/docs/query/overview) queries are detected as SQL queries and therefore raise `S608` violations.

Example:
```python
query = f"""
    SELECT
    customer.status
    FROM customer
    WHERE customer.id = {customer_id}
"""
```

What would be the best way to handle this without turning off S608 for this specific piece of code?

Further information about GAQL and escaping: https://github.com/googleads/google-ads-python/issues/732

**Comment by lociii** [comment]
> What happens if the query is changed so that it contains a syntax error? What if it is changed to use more resources than necessary? What if it is changed so that extra data is returned?
> The comment in the linked issue just says "[GAQL is] not susceptible to the same types of injection attacks as SQL". I don't think that is the same as "GAQL is completely fool-proof".

I guess Google doesn't care as it's limited on their side by the scope of the API token. When managing multiple customers, it's the responsibility of the managing account to make sure no data is leaked. \
So yes, some kind of protection is needed.

> You can use `per-file-ignores` to disable `S608` on a selected set of files if you know those fails don't contain any "real" SQL statements.

That's what we're currently doing.

Alright. Thanks!
Just wanted to make sure that we're not missing anything.
